### ISSUE TITLE
Update capability to replace arkade with a newer version

### ISSUE THREAD

**Issue by alexellis** [alexellis/arkade#973]
It would be possible to self-update by running `arkade update` and having the running binary replaced by a newer version.

`arkade get arkade` has the logic to download the latest version, but we need something to replace this version.

I'd also like to see the self-check only download the binary if there's a newer version, something which we can find a way to optimise.

This way, people could add `arkade update` to their .bashrc, profile, or a daily cron.

## Expected Behaviour

Update command can replace the running version

## Current Behaviour

It prints out install instructions, which can be copy/pasted

## Possible Solution

* Use the equivalent code of `arkade get arkade` to get an updated version
* Move that file to replace the location of the currently running binary
* Make sure this only downloads the ~ 10MB binary, if it's required

## Context

I've been seeing a lot more breakages recently with projects changing binary names and formats, this would make arkade self-update, and work better for when we've fixed issues introduced by upstream projects.

**Comment by alexellis** [comment]
@Jasstkn / @mrwormhole do you have time for a challenge?

**Comment by mrwormhole** [comment]
yes makes sense to me, is it potentially safe to swap old arkade binary with new arkade binary during the old arkade CLI process? I believe it is safe on linux, but on windows I have doubts

**Comment by alexellis** [comment]
Perhaps we'll start with Linux first then?

**Comment by Jasstkn** [comment]
Hm, on Mac I just have an alias to do the following:
```
which upd_arkade
upd_arkade: aliased to sudo curl -sLS https://get.arkade.dev | sudo sh
```

**Comment by alexellis** [comment]
The first sudo shouldn't be needed for `curl`

**Comment by Jasstkn** [comment]
> The first sudo shouldn't be needed for `curl`

yep, you're right!

**Comment by Shikachuu** [comment]
How about we get the current binary first with [`os.Executable()`](https://pkg.go.dev/os@master#Executable) then check in the get command if we are updating `arkade` itself and in that case we set the `movePath` to the result?
Also everything in the `get` package seems path aware, so we should approach this with `get.Download()` like everywhere else.

Honestly I don't know if this will work on windows, but I'm quite sure it this will solve the problem for our unix friends.

**Comment by Jasstkn** [comment]
I have a windows to test on now 😃

**Comment by Shikachuu** [comment]
> How about we get the current binary first with [`os.Executable()`](https://pkg.go.dev/os@master#Executable) then check in the get command if we are updating `arkade` itself and in that case we set the `movePath` to the result? Also everything in the `get` package seems path aware, so we should approach this with `get.Download()` like everywhere else.
> 
> Honestly I don't know if this will work on windows, but I'm quite sure it this will solve the problem for our unix friends.

@alexellis, @Jasstkn is this solution seems good for you guys?

I can start to skeetch up a PR this weekend or so.
